codec-http/src/main/java/io/netty/handler/codec/http/multipart/DeleteFileOnExitHook.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http.multipart;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * DeleteFileOnExitHook.
+ */
+final class DeleteFileOnExitHook {
+    private static final Set<String> FILES = Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
+
+    private DeleteFileOnExitHook() {
+    }
+
+    static {
+        // DeleteOnExitHook must be the last shutdown hook to be invoked.
+        // Application shutdown hooks may add the first file to the
+        // delete on exit list and cause the DeleteOnExitHook to be
+        // registered during shutdown in progress.
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+
+            @Override
+            public void run() {
+                runHooks();
```

Comment:
run deletion hook?

Reply:
😵，  Change `runHooks` method name?

Reply:
right, eg. runDeletionHook, plural hooks doesn't make sense because it is only one hook.

Reply:
Good advice ! 😊

Reply:
did you forgot to push ?

Reply:
> did you forgot to push ?

⊙﹏⊙‖∣     

 Done.